tools/autograd/gen_autograd_functions.py

```diff
@@ -4,11 +4,17 @@
 #  Functions.h/cpp: subclasses of autograd::Node
 #  python_functions.h/cpp: Python bindings for the above classes
 #
-import os
 import re
-from .utils import nested_dict, CodeTemplate, write
 from .gen_autograd import VIEW_FUNCTIONS
-from .utils import IDENT_REGEX
+
+from typing import List, Sequence, Tuple, Optional
+
+from tools.codegen.api.autograd import *
```

Comment:
nit: you always import the codegen part with `*` what is the reason here? it makes quite a few names below hard to track down where they come from :/

Reply:
I can add explicit imports - @ezyang shall we make it a convention?